src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todo, setTodo] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+
+  const [isUserErrored, setUserErrored] = useState(false);
+  const [isTitleErrored, setTitleErrored] = useState(false);
+
+  const handleTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setTitleErrored(false);
+  };
+
+  const handleUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setUser(+e.target.value);
+    setUserErrored(false);
+  };
+
+  const addTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const titleTrim = title.trim();
+
+    setTitleErrored(!titleTrim);
+
+    setUserErrored(!user);
+
+    if (titleTrim && user) {
```

Comment:
You could just add return to an opposite condition to avoid nesting
```suggestion
    if (!titleTrim || !user) {
      return
    }
    
    const newTodo = { ...
```